ruby_enumerable_methods.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    select_arr = []
+    for item in self do
+      select_arr << item if yield(item)
+    end
+    select_arr
+  end
+
+  def my_all?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break unless pass
+    end
+    pass
+  end
+
+  def my_any?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break if pass
+    end
+    pass
+  end
+
+  def my_none?
+    pass = false
+    for item in self do
+      pass = !yield(item)
+      break unless pass
+    end
+    pass
+  end
```

Comment:
- enumerables my_none? when no block or argument is given returns true only if none of the collection members is true
      Failure/Error: expect(false_array.my_none?).to be true

- enumerables my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be true

-  enumerables my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
      
- enumerables my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)